helm/external-dns-app/templates/serviceaccount.yaml

```diff
@@ -6,6 +6,9 @@ metadata:
   {{- if and (eq .Values.provider "aws") (eq .Values.aws.irsa "true") (eq .Values.aws.access "internal") }}
   annotations:
     eks.amazonaws.com/role-arn: "arn:aws:iam::{{ .Values.aws.accountID }}:role/{{ template "aws.iam.role" . }}"
+  {{- else if and (eq .Values.provider "gcp") }}
```

Comment:
Maybe you also want to check here if `.Values.gcpProject` is set?

Reply:
Yeah. Take a look now.